sklearn/decomposition/_dict_learning.py

```diff
@@ -906,32 +913,28 @@ def transform(self, X):
 
         Returns
         -------
-        X_new : array, shape (n_samples, n_components)
+        X_new : array of shape (n_samples, n_components)
             Transformed data
-
         """
         check_is_fitted(self)
+        return self._transform(X, self.components_)
 
-        X = check_array(X)
-
-        code = sparse_encode(
-            X, self.components_, algorithm=self.transform_algorithm,
-            n_nonzero_coefs=self.transform_n_nonzero_coefs,
-            alpha=self.transform_alpha, max_iter=self.transform_max_iter,
-            n_jobs=self.n_jobs, positive=self.positive_code)
-
-        if self.split_sign:
-            # feature vector is split into a positive and negative side
-            n_samples, n_features = code.shape
-            split_code = np.empty((n_samples, 2 * n_features))
-            split_code[:, :n_features] = np.maximum(code, 0)
-            split_code[:, n_features:] = -np.minimum(code, 0)
-            code = split_code
 
-        return code
+class _BaseSparseCoding:
+    """Base class to set shared parameters of sparse coding classes."""
+    def __init__(self, transform_algorithm, transform_n_nonzero_coefs,
+                 transform_alpha, split_sign, n_jobs, positive_code,
+                 transform_max_iter):
+        self.transform_algorithm = transform_algorithm
```

Comment:
Is having this abstraction desirable just for initializing parameters? The alternative is to list everything in the child's `__init__` methods.

Reply:
Actually, we could remove the mixin and just move the `transform` and `_transform` in the base class.
WDYT?

Reply:
Sounds good to me!

Reply:
Done!